zerver/lib/outgoing_webhook.py

```diff
@@ -62,6 +63,10 @@ def make_request(self, base_url: str, event: dict[str, Any], realm: Realm) -> Re
             apply_markdown=False,
             client_gravatar=False,
             keep_rendered_content=True,
+            can_access_sender=check_can_access_user(
+                get_user_profile_by_id(event["message"]["sender_id"]), self.user_profile
```

Comment:
In theory we could optimize it by checking `check_user_can_access_all_users(self.user_profile)` before querying the user. Not sure it's worth the effort; it's part of a somewhat hot code path but also a very fast check compared to sending the message in the first place.

Reply:
- #32585